internal/controller/numaresourcesscheduler_controller.go

```diff
@@ -224,7 +226,11 @@ func (r *NUMAResourcesSchedulerReconciler) syncNUMASchedulerResources(ctx contex
 	klog.V(4).Info("SchedulerSync start")
 	defer klog.V(4).Info("SchedulerSync stop")
 
-	schedSpec := instance.Spec.Normalize()
+	//intentionally define the feature gate here instead of pulling it because some verions may not have it supported at all
+	KubeletPodResourcesListUseActivePods := "KubeletPodResourcesListUseActivePods"
+	useActivePodsOnly := utilfeature.DefaultFeatureGate.Enabled(featuregate.Feature(KubeletPodResourcesListUseActivePods))
+
+	schedSpec := instance.Spec.Normalize(useActivePodsOnly)
```

Comment:
not really. This is well-intentioned but doesn't work.
The problem is now you import the packages from the version we depend on, and you consume that value. This means the default value hardcoded in the libraries we import and you use here will be the truth value. But this is not what we want. We want to change the default depending on what is actually on the cluster.

So what we should do, I think, is rather

- detect the full cluster version (X.Y.Z) and pass this information to the `numresourcesschedulereconcciler` object.
- keep the normalization stage as-is
- add a pre-normalization stage. IFF the user did NOT set a value, compute the value based on the cluster version
- call the API normalization (which is static and must not depend on anything else) as usual

Reply:
the only way is depending on the version once we have a fixed-version in the kubelet.

Reply:
clarified offline